test/logics/radiant-v2/deposit.test.ts

```diff
@@ -0,0 +1,89 @@
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import { claimToken, getChainId, snapshotAndRevertEach } from '@protocolink/test-helpers';
+import * as common from '@protocolink/common';
+import * as core from '@protocolink/core';
+import { expect } from 'chai';
+import hre from 'hardhat';
+import * as radiantv2 from 'src/logics/radiant-v2';
+import * as utils from 'test/utils';
+
+describe('arbitrum: Test RadiantV2 Deposit Logic', () => {
+  let chainId: number;
+  let user: SignerWithAddress;
+
+  before(async () => {
+    chainId = await getChainId();
+    [, user] = await hre.ethers.getSigners();
+    await claimToken(chainId, user.address, radiantv2.arbitrumTokens.USDC, '100');
+    await claimToken(chainId, user.address, radiantv2.arbitrumTokens.WETH, '100');
+  });
+
+  snapshotAndRevertEach();
+
+  const testCases = [
+    {
+      input: new common.TokenAmount(radiantv2.arbitrumTokens.ETH, '1'),
+      tokenOut: radiantv2.arbitrumTokens.rWETH,
+    },
+    {
+      input: new common.TokenAmount(radiantv2.arbitrumTokens.WETH, '1'),
+      tokenOut: radiantv2.arbitrumTokens.rWETH,
+    },
+    {
+      input: new common.TokenAmount(radiantv2.arbitrumTokens.USDC, '1'),
+      tokenOut: radiantv2.arbitrumTokens.rUSDC,
+    },
+    {
+      input: new common.TokenAmount(radiantv2.arbitrumTokens.ETH, '1'),
+      tokenOut: radiantv2.arbitrumTokens.rWETH,
+      balanceBps: 5000,
+    },
+    {
+      input: new common.TokenAmount(radiantv2.arbitrumTokens.WETH, '1'),
+      tokenOut: radiantv2.arbitrumTokens.rWETH,
+      balanceBps: 5000,
+    },
+    {
+      input: new common.TokenAmount(radiantv2.arbitrumTokens.USDC, '1'),
+      tokenOut: radiantv2.arbitrumTokens.rUSDC,
+      balanceBps: 5000,
+    },
+  ];
+
+  testCases.forEach(({ input, tokenOut, balanceBps }, i) => {
+    it(`case ${i + 1}`, async () => {
+      // 1. get output
+      const radiantV2DepositLogic = new radiantv2.DepositLogic(chainId);
+      const { output } = await radiantV2DepositLogic.quote({ input, tokenOut });
+
+      // 2. build funds, tokensReturn
+      const tokensReturn = [output.token.elasticAddress];
+      const funds = new common.TokenAmounts();
+      if (balanceBps) {
+        funds.add(utils.calcRequiredAmountByBalanceBps(input, balanceBps));
+        tokensReturn.push(input.token.elasticAddress);
+      } else {
+        funds.add(input);
+      }
+
+      // 3. build router logics
+      const routerLogics: core.IParam.LogicStruct[] = [];
```

Comment:
Suggest update `IParam` since the name has been changed on contract side

Reply:
Could you give me more information about this? I currently not familiar with the code on contract side

Reply:
updated.